### PR TITLE
white logo on map and hover

### PR DESCRIPTION
**Pull request by Buesi**
- added new white mapzy logo 
- changes to normal logo on hover
- changed all logos to sidetext (more consistent)

Fixed some additional small issues:
- decreased width of "add location" button for small screens to prevent overlap with mapzy logo
- set margin top to 24 on login page only for 2xl screens to prevent last sentence (log in if you already have an account) being cut off

### CODE REVIEW

**Review by sagzy — COMMENTED** [review]
Hey @Buesi! Congrats on your first PR 🎉

Some comments:
1. The font of the logo does not load for me (see screenshots). It's probably due to the SVG, some solution can be found [here](https://graphicdesign.stackexchange.com/questions/10733/how-do-i-use-a-custom-font-in-an-svg-image-on-my-site), or alternative we can use PNGs. Given that the image are going to be small in size, the image size in png should also be acceptable
2. The hover effect doesn't always work for me. Maybe also linked to the SVG? Does it always work for you? Here's an alternative [implementation](https://www.tutorialrepublic.com/faq/how-to-change-image-on-hover-with-css.php) as reference
3. Can we still reduce a bit the size of the overlay logo? I still find it a bit too apparent, especially on a big screen
4. It's an illusion I know, but I always find the logo with side text "unaligned". I'd personally prefer to keep the logo with the text below in the top bar at least, as we do on the [landing page](https://mapzy.io/)
5. Bonus: on mobile, if the little white thing from Mapbox is expanded, it conflicts with the overlay image (see screenshots). Maybe we can play with the z-index so that it doesn't?


Screenshots:
Screenshot 2022-03-08 at 21 03 38
Screenshot 2022-03-08 at 21 03 32
Screenshot 2022-03-08 at 20 53 42
Screenshot 2022-03-08 at 21 24 38

**Comment by Buesi** [comment]
>     1. The font of the logo does not load for me (see screenshots). It's probably due to the SVG, some solution can be found [here](https://graphicdesign.stackexchange.com/questions/10733/how-do-i-use-a-custom-font-in-an-svg-image-on-my-site), or alternative we can use PNGs. Given that the overlay image is going to be small, the image size shouldn't be a problem, even with PNG
> 
should be fixed now with the .png files

>     2. The hover effect doesn't always work for me. Maybe also linked to the SVG? Does it always work for you? Here's an alternative [implementation](https://www.tutorialrepublic.com/faq/how-to-change-image-on-hover-with-css.php) as reference
> 
It always works for me, if it still doesn't work for you, can you tell me in what case? after you clicked somewhere else or just random working, not working and working again?
>     3. Can we still reduce a bit the size of the overlay logo? I still find it a bit too apparent, especially on a big screen
> 
done, let me know if you want it even smaller, in my opinion, if we set it even lower to 1.4, the readability suffers.
>     4. It's an illusion I know, but I always find the logo with side text "unaligned". :D I'd personally prefer to keep the logo with the text below in the dashboard menu bar, as we do on the [landing page](https://mapzy.io/)
> 
changed the logo a bit, let me know if you still prefer the stacked logo
>     5. Bonus: on mobile, if the little white thing from Mapbox is expanded, it conflicts with the overlay image (see screenshots). Maybe we can play with the z-index so that it doesn't?
> 
i think setting the z-value of the mapzy logo to 0 solved that problem (see pic), does it look like what you wanted?
> 
Screenshot 2022-03-08 at 21 24 38

Screenshot 2022-03-09 at 14 19 03

**Comment by shafy** [comment]
Can't we use vectorized SVG so it's not dependent on a font? If PNG is easier, can you please make sure that the width is max 2x the final width displayed (so if the width on the map is 100px, the width of the file should be 200px). And also to run it thorugh a compressor. Like this, the file sizes are too large.

And regarding hover effect: I think @sagzy means that it's better to use the technique where you set the image as the background image of a div and then use the `:hover` attribute to change the background image in CSS. At least, that's how it's usually done and may solve the problem that currently it sometimes doesn't work anymore. This could be due to one of the two overlaying the other even if the opacity is 0, and then the hover doesn't work anymore.

**Comment by Buesi** [comment]
Ok, I'll try that!

**Comment by Buesi** [comment]
@sagzy just wanted to make sure you saw my new latest :)

File: app/views/shared/_map.html.erb [hunk 1]
```diff
@@ -17,7 +17,9 @@
       </turbo-frame>
     </div>
     <%= render "shared/mapbox" %>
-    <%= link_to "https://mapzy.io/?source=embedded", class: "absolute bottom-2 left-2 z-10", target: "_blank" do %>
-      <%= image_tag "mapzy_logo_text_side.png", class: "h-[1.7rem]" %>
+
+    <%= link_to "https://mapzy.io/?source=embedded", class: "absolute bottom-2 left-2 z-0", target: "_blank" do %>
+      <div class="h-[1.4rem] w-24 md:h-[1.6rem] bg-contain bg-no-repeat bg-[url('logo_white_border_sidetext_transparent.svg')] hover:bg-[url('logo_sidetext_transparent.svg')]">
+      </div>
```
Comment (sagzy) [inline app/views/shared/_map.html.erb:23]: The reason why the hover effect does not always work I think is: the <a> container wraps the <div> container, which has the background images. I think if you flip the order of these two, it should work better:

```
  <div class="bg-... hover:bg-..." ...>
     <a></a>
   </div>
```

**Review by sagzy — COMMENTED** [review]
Thanks a lot for the changes @Buesi! Great stuff, the logo with the side text is much more aligned visually, the z-index trick works well and there is no more font issue. 👏

Three additional comments:
1. The hover effect does not work for me all the time. But I think it's now just due to an order issue. See my comment on the `/shared/_map.html.erb` file

https://user-images.githubusercontent.com/6225080/158263710-2605a622-f397-4dc6-b577-6b98960d0e93.mov

2. On mobile, the mapzy logo in the dashboard is too prominent. Could you replace it with a logo only version (no text)?

Screenshot 2022-03-14 at 22 18 03

3. We can now delete the unused png logos

Bonus. The security check is not passing, due to a dependency that needs to be upgraded. Would you like to take a look at it or shall I?

**Comment by shafy** [comment]
> On mobile, the mapzy logo in the dashboard is too prominent. Could you replace it with a logo only version (no text)?

@sagzy I think we should leave it. Because most mobile usage will not be in the dashboard (rather on the embedded map), and I think there it's important that visitors see the actual name "Mapzy". And making two different versions for dashboard vs. embedded seems a bit overengineered for now. What do you think?

**Comment by sagzy** [comment]
> > On mobile, the mapzy logo in the dashboard is too prominent. Could you replace it with a logo only version (no text)?
> 
> @sagzy I think we should leave it. Because most mobile usage will not be in the dashboard (rather on the embedded map), and I think there it's important that visitors see the actual name "Mapzy". And making two different versions for dashboard vs. embedded seems a bit overengineered for now. What do you think?

I meant in the dashbaord header (the big mapzy top left), not the logo on the map. It used to be the logo with text at the bottom here and would work better visually with the mobile layout. As we've replaced it now with text on the side, it takes too much space on mobile imo. So would prefer the logo without text, or put the logo with the text below in that case. I agree not fundamental, but still quick win

**Comment by sagzy** [comment]
> > > On mobile, the mapzy logo in the dashboard is too prominent. Could you replace it with a logo only version (no text)?
> > 
> > 
> > @sagzy I think we should leave it. Because most mobile usage will not be in the dashboard (rather on the embedded map), and I think there it's important that visitors see the actual name "Mapzy". And making two different versions for dashboard vs. embedded seems a bit overengineered for now. What do you think?
> 
> I meant in the dashbaord header (the big mapzy logo top left), not the logo on the map. It used to be the logo with text at the bottom here and would work better visually with the mobile layout. As we've replaced it now with text on the side, it takes too much space on mobile imo. So would prefer the logo without text, or put the logo with the text below in that case. I agree not fundamental, but still quick win

Or just put the logo with text below all the time in the dashboard header

**Comment by shafy** [comment]
@sagzy Oh I see, sorry thought you mean the other logo. Yes, or we can just make it smaller for mobile? I think @Buesi's point that it's weird to see two different versions of the logo on one screen is valid.

**Comment by sagzy** [comment]
What about on the landing page? https://mapzy.io/ 

To me it looks okay the current way, the text below in the header & a different one for the embedded map.

**Comment by shafy** [comment]
> What about on the landing page? https://mapzy.io/
> 
> To me it looks okay the current way, the text below in the header & a different one for the embedded map.

Landing page we can leave it, because if you don't have a big screen you won't see it at the same time anyways. But I don't have a very strong opinion on this. I just trust @Buesi because she is usually good with visual stuff like this and I suck at it. You can have the final word @sagzy

**Comment by sagzy** [comment]
> > What about on the landing page? https://mapzy.io/
> > To me it looks okay the current way, the text below in the header & a different one for the embedded map.
> 
> Landing page we can leave it, because if you don't have a big screen you won't see it at the same time anyways. But I don't have a very strong opinion on this. I just trust @Buesi because she is usually good with visual stuff like this and I suck at it. You can have the final word @sagzy

I agree, @Buesi has the eye. :D

My preferred solution would be:
- keep the logo with text below in every header (landing page, the app, mobile & web). Like it is today
- on the map, show the logo only when the map is embedded, but not on the dashboard anymore. There is actually no need to spam the dashboard with the overlay logo, it's meant to advertise it Mapzy once it's embedded. This way we also don't show two styles of logo on the dashboard (which I agree looks a bit odd).

What do you think of this solution @Buesi?

**Comment by shafy** [comment]
Fine with me. I thought about this actually before, but didn't mention it because it makes the map less WYSIWYG. But I think that's fine for the sake of improved UX.

**Comment by Buesi** [comment]
I think that's a great solution!

**Comment by Buesi** [comment]
finally finished this :)

- hover should now work reliably
- mapzy logo only on embedded maps
- put stacked mapzy logo back everywhere

**Comment by sagzy** [comment]
It's perfect! Thanks a lot @Buesi. I'll merge the PR and release your changes to production :)

**Review by sagzy — APPROVED** [review]

**Pull request merged**